test/test.js

```diff
@@ -669,6 +670,38 @@ describe('Routes', () => {
     })
   })
 
+  describe('Redirect', () => {
+    it('redirects to the right domain', async () => {
+      let server
+      try {
+        const handler = await createHandler({
+          client,
+          logger: {
+            info () {},
+            error (...args) { console.error(...args) }
+          },
+          async getCurrentRound () {
+            return currentSparkRoundNumber
+          },
+          domain: 'foobar'
+        })
+        server = http.createServer(handler)
+        server.listen()
+        await once(server, 'listening')
+        const spark = `http://127.0.0.1:${server.address().port}`
+        const res = await fetch(
+          `${spark}/rounds/${currentSparkRoundNumber}`,
+          { redirect: 'manual' }
+        )
+        await assertResponseStatus(res, 301)
+        assert.strictEqual(res.headers.get('location'), `https://foobar/rounds/${currentSparkRoundNumber}`)
+      } finally {
+        server.closeAllConnections()
+        server.close()
```

Comment:
It makes me wonder: instead of closing the server at the end, would it be enough to call `server.unref()` right after we create it?